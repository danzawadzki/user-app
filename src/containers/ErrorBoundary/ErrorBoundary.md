Default ErrorBoundary with negative flag (higher order component):

```jsx
<ErrorBoundary hasError={false}>Test</Module>
```

Default ErrorBoundary with positive flag (higher order component):

```jsx
<ErrorBoundary hasError={true}>Test</Module>
```
