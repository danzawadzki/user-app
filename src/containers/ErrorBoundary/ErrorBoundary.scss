@import '../../assets/styles/colors';

.ErrorBoundary {
	&__heading {
		color: $brand--accent;
		text-align: center;
		font-size: 3rem;
		margin: 1rem;
	}

	&__subheading {
		color: $brand--main;
		text-align: center;
	}
}
