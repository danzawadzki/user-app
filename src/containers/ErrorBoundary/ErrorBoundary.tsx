import * as React from 'react';
import { Component } from 'react';
import './ErrorBoundary.scss';

/** Interface for accordion props */
export interface IErrorBoundaryProps {
	/** Error flag */
	hasError: boolean;
	/** Body content */
	children: JSX.Element | JSX.Element[] | string;
}

/** Interface for accordion props */
export interface IErrorBoundaryState {
	/** Error flag */
	hasError?: boolean;
}

/**
 * Error Boundary component
 *
 * @version 1.0.0
 */
export default class ErrorBoundary extends Component<
	IErrorBoundaryProps,
	IErrorBoundaryState
> {
	state = { hasError: false };

	componentDidCatch() {
		this.setState({
			hasError: true
		});
	}

	render() {
		if (this.props.hasError || this.state.hasError) {
			return (
				<div className="ErrorBoundary">
					<h3 className="ErrorBoundary__heading">Ooooops!</h3>
					<span className="ErrorBoundary__subheading">
						Something went wrong! Make sure the server is working properly.
					</span>
				</div>
			);
		} else {
			return this.props.children;
		}
	}
}
