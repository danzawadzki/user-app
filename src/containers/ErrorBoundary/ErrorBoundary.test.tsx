import React from 'react';
import { shallow } from 'enzyme';
import ErrorBoundary from './ErrorBoundary';

describe('ErrorBoundary component: ', () => {
	// Shallow rendered component
	const content = <p className="test">Test</p>;

	const component = shallow(<ErrorBoundary hasError={false}>{content}</ErrorBoundary>);

	it('should render match snapshot', () => {
		expect(component).toMatchSnapshot();
	});

	it('should render correctly', () => {
		expect(component.find('p.test')).toBeDefined();
	});

	it('should render with child component', () => {
		expect(component.html()).not.toBe(null);
		expect(component.find(content)).toBeTruthy();
	});

	it('should render error message', () => {
		const component = shallow(
			<ErrorBoundary hasError={true}>{content}</ErrorBoundary>
		);

		expect(component.find(content)).not.toExist();
		expect(component.find('h3.ErrorBoundary__heading')).toBeTruthy();
	});
});
