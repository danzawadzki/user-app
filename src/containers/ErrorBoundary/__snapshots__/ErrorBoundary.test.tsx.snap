// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary component:  should render match snapshot 1`] = `
<p
  className="test"
>
  Test
</p>
`;
